File: web-spring/src/main/java/io/github/agussyahrilmubarok/web/service/IAuthService.java
package io.github.agussyahrilmubarok.web.service;

import io.github.agussyahrilmubarok.web.model.auth.SignUpRequest;
import io.github.agussyahrilmubarok.web.model.user.UserResponse;

public interface IAuthService {

    UserResponse signUp(final SignUpRequest request);
}

File: examples/web-mvc-i8n/web-dotnet/Web/Application/Service/IAuthService.cs
using Web.Application.DTO.Auth;
using Web.Application.DTO.User;

namespace Web.Application.Service;

public interface IAuthService
{
    Task<UserResponse> SignUpAsync(SignUpRequest request, CancellationToken ct = default);
    Task<UserResponse> SignInAsync(SignInRequest request, CancellationToken ct = default);
}